File: Codechef/Food_Plan/Food_Plan.cpp
#include <iostream>
using namespace std;

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        float N, M, least;
        cin >> N >> M;
        N -= 0.1*N;
        least = min(N, M);

        if (M == N)
        {
            cout << "EITHER" << endl;
        }
        else if (least == N)
        {
            cout << "ONLINE" << endl;
        }
        else if (least == M)
        {
            cout << "DINING" << endl;
        }

    }
    return 0;
}

File: Codechef/Food_Plan/Food_Plan.py
T = int(input())
while T:
    T -= 1
    N, M = map(int, input().split())
    N -= 0.1*N
    least = min(M, N)
    if least == N:
        print("ONLINE")
    elif least == M:
        print("DINING")
    if M == N:
        print("EITHER") 